### DISCUSSION TITLE
Loading Component support slot

### DISCUSSION THREAD

**Discussion by microkof** [element-plus/element-plus#987] [Ideas]
In modern front-end development, if there is an interface error, the best practice is to change the rotation SVG of loading into the error SVG, and put a "重新加载" button below.

I Hope Loading Component to support slots function.

I know that 2.0 version doesn't have this capability. I hope the plus version can add it.

Thank you!

Thank developers!

**Comment by xiaoxian521** [comment]
其实我认为那样的话 会增加用户的操作 是不友好的 
你可以尝试 当检查到加载失败，采用倒计时方式自动 重新加载 这样效果是否会好一点呢

**Comment by microkof** [comment]
官方删了我的Issue，让我在这发，发了又没人搭理。
我果然在开源社区生存艰难。

**Reply by jw-foss** [reply]
Hey, it's not personal, we are not full time contributors to this project. None of us is. @xiaoxian521 has given advice that you can try on. Note that, discussion is meant to discuss, instead of making promises, loading should only indicate the state of the current data fetching process, instead of adding extra interaction to it.
BTW, I didn't **DELETE** your issue, I **CLOSED** it, because we want to keep everything tracked.

Also, open a discussion doesn't mean that this should be implemented, we value every user's opinion about this whole UI.

**Reply by jw-foss** [reply]
Although I think the `slot` feature is worthy implementing refer to the popular [AntDesign](https://ant.design/components/spin/), they have allowed user to pass slots to the `Spin` component, which I think it's great, because users sometimes don't need the `v-if` `v-else` branches to control that.

**Reply by jw-foss** [reply]
I value your request, but we also need resource to do that. Right now as you can see that we haven't been able to publish the first official release yet, still in `Beta`, there were bugs to be fixed, features to be implemented. The name of this project is Element**Plus**, the more important part is all about the `Plus`. We want to make this project to thrive. That's why I asked you to put this discussion here. Processing these documentation will cost time which we currently don't have that much.

Hope that you understand, even if we replied this right away, it would still be on hold for a period of time.

**Reply by microkof** [reply]
> I value your request, but we also need resource to do that. Right now as you can see that we haven't been able to publish the first official release yet, still in `Beta`, there were bugs to be fixed, features to be implemented. The name of this project is Element**Plus**, the more important part is all about the `Plus`. We want to make this project to thrive. That's why I asked you to put this discussion here. Processing these documentation will cost time which we currently don't have that much.
> 
> Hope that you understand, even if we replied this right away, it would still be on hold for a period of time.

哥，你打了这么多字，别激动。
我就是试试有没有人回复我。
别的没啥。
我代表了一批程序员的想法，就是，当我提个建议，希望官方说“good idea”，或者“不予采纳”，就行了，什么时候能实现无所谓，开发总有优先级。
忙先，勿回。

@xiaoxian521 没谁家用倒计时然后自动重连的。充其量会静默定时重连，目前我家采用的就是这个办法，只不过这也不是好办法。现在2020年，服务器出错的概率如果是1个单位，那么网络出错概率可能是200个单位，所以我更多考虑服务于网络出错。当UI给用户提示“网络错误”之后，用户只需看提示、调试网络、然后自主点击“重新加载”按钮就好了，而不是只能盯着倒计时。

**Comment by xiaoxian521** [comment]
....emm 官方讲的很清楚啊  所有需求在这里谈论

**Comment by szulcus** [comment]
any updates?
